common-content/en/module/logic/search/index.md

```diff
@@ -0,0 +1,119 @@
+> 🐀 You have **500** rats you can use to test the wine.  
+> ⌛ The poison is lethal and will kill within **1** hour.
+
+{{<tabs name="boolean">}}
+===[[Write down your solution]]===
+
+In your notebook, write down your solution to this problem. Here are the constraints again:
+
+🍾 You have **1000** wine bottles, one of which is poisoned.  
+🐀 You have **500** rats you can use to test the wine.  
+⌛ The poison is lethal and will kill within **1** hour.
+
+===[[🍾 🍾 🐀 Look at a solution]]===
+
+### 🍾 🍾 🐀
+
+One way to think of this is to reduce this problem to its simplest case. If you have 2 bottles and 1 rat, you can test the wine by giving the rat a sip from one bottle. Wait one hour. If the rat is alive, the bottle is safe. If the rat is dead, the bottle is poisoned.
+
+With 500 rats, it's similar, but not the same. Divide the 1000 bottles into 500 groups of 2. In round 1, give each rat a sip from one bottle in each group. Wait one hour. If the rat is alive, the bottle is safe. If the rat is dead, you found the poisoned bottle. If all rats are alive, the poisoned bottle is in the 500 bottles that were not tested. You still have 500 rats, so you can have each rat test the other bottle in their group.
+
+This solution will take a maximum of 2 hours.
+
+{{</tabs>}}
+
+> #### New condition
+>
+> You now have just **10** rats to test the wine.
+
+{{<tabs name="decimation">}}
+===[[Write down your solution]]===
+
+In your notebook, write down your solution to this problem. Here are the constraints again:
+
+🍾 You have **1000** wine bottles, one of which is poisoned.  
+🐀 You have **10** rats you can use to test the wine.  
+⌛ The poison is lethal and will kill within **1** hour.
+
+===[[Look at a solution]]===
+
+With just 10 rats, we need a way to make our problem simpler. Think of it as a search pattern, narrowing down our suspects with each round of testing.
+
+**Round 1:** Let's divide our 1000 bottles into 10 groups of 100. Each rat tests one group. An hour later, with one dead rat, we have narrowed our search to 100 bottles. We've saved 900 bottles of wine and we still have 9 rats to help us.
+
+**Round 2** Let's do it again: take our suspicious 100 bottles and create ten groups of ten. Our doomed rat perishes again, and after another hour, we've found the right group of ten. We now have 8 rats.
+
+**Round 3** Now we're down to just 10 bottles. Eight rats test a bottle each, leaving two untested. If a rat dies, we've found the poisoned bottle. If not, we have two bottles left for the last round.
+
+**Round 4: Base case** We can use just one rat to test one bottle. If we have a dead rat, it's the poisoned bottle. If not, the last untasted bottle is the culprit.
+
+Through careful division and patience, we've turned an overwhelming search into a manageable investigation. Four hours, and we'll have our answer.
+
+{{</tabs>}}
+
+> #### New condition
+>
+> The birthday party is now in **1** hour, so you only have time for one test!
```

Comment:
I think we should frame this slightly differently - we're really asking a different question here.

My initial question was: How many rats do we have in this iteration?

But we're not playing the same game - now we're playing a game of "how many rats would we need" rather than "how do we use the number of rats we have" - let's frame it that way, rather than saying "Same game: New condition" which I think we're currently doing?

Reply:
I think it's the same game as I'm still giving them 10 rats. They only need 7 in the end, which we can hammer down on afterwards? I only want to add one more constraint, not completely change the game, to provide a bridge from the last approach.